Mathlib/Tactic/Linter/Style.lean

```diff
@@ -466,4 +468,50 @@ initialize addLinter doubleUnderscore
 
 end Style.nameCheck
 
+/-! # The "openClassical" linter -/
+
+/-- The "openClassical" linter emits a warning on `open Classical` statements which are not
+scoped to a single declaration. A non-scoped `open Classical` can hide that some theorem statements
+would be better stated with explicit decidability statements.
+-/
+register_option linter.style.openClassical : Bool := {
+  defValue := true
+  descr := "enable the openClassical linter"
+}
+
+namespace Style.openClassical
+
+/-- If `stx` is syntax describing an `open` command, `extractOpenNames stx`
+returns an array of the syntax corresponding to the opened names,
+omitting any renamed or hidden items. -/
+def extractOpenNames : Syntax → Array Syntax
+  | `(command|open $arg hiding $_*)    => #[arg]
+  | `(command|open $arg renaming $_,*) => #[arg]
+  | `(command|open $arg ($_*))         => #[arg]  -- `openOnly`, hence the parens in `($_*)`
+  | `(command|open $args*)             => args
+  | `(command|open scoped $args*)      => args
+  | _ => #[]
+
+@[inherit_doc Mathlib.Linter.linter.style.openClassical]
+def openClassicalLinter : Linter where run := withSetOptionIn fun stx ↦ do
+    unless Linter.getLinterValue linter.style.openClassical (← getOptions) do
+      return
+    if (← MonadState.get).messages.hasErrors then
+     return
+    -- TODO: once mathlib's Lean version includes leanprover/lean4#4741, make this configurable
+    unless #[`Mathlib, `test, `Archive, `Counterexamples].contains (← getMainModule).getRoot do
+      return
+    -- If `stx` describes an `open` command, extract the list of opened namespaces.
+    for stxN in (extractOpenNames stx).filter (·.getId == `Classical) do
+      Linter.logLint linter.style.openClassical stxN "\
+      please avoid 'open (scoped) Classical' statements: this can hide theorem statements\n\
```

Comment:
```suggestion
      please avoid 'open (scoped) Classical' statements: this can hide theorem statements \
```
I think that it is better to leave the formatting to the pretty-printer-